app/javascript/components/widgets/weather/weather-effect.jsx

```diff
@@ -0,0 +1,386 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css, keyframes } from 'styled-components';
+import gql from 'graphql-tag';
+import withFragment from '@hocs/with-fragment';
+
+const getWeatherEffect = gql`
+  fragment WeatherEffect on Weather {
+    current {
+      weather {
+        main
+        id
+      }
+    }
+  }
+`;
+
+const EffectContainer = styled.div`
+  position: absolute;
+  top: 0px;
+  left: 0px;
+  right: 0px;
+  bottom: 0px;
+  mask-image: radial-gradient(farthest-side at bottom, #000e, #000b 70%, #0000);
+  //mask-image: linear-gradient(to right,#0000,#000b,#000e,#000e,#000b,#0000);
+`;
+
+const Drop = styled.div`
+  position: absolute;
+  bottom: 100%;
+  pointer-events: none;
+`;
+
+const floorTo = (value, decimal) => {
+  const exp = 10 ** decimal;
+  return Math.floor(value * exp) / exp;
+};
+
+const randomDistributeWithSpacing = (index, total) => {
+  return floorTo(((index + Math.random() * 0.8) * 100) / total, 2);
```

Comment:
The idea is to evenly distribute positions out of a total. Such as for every position between 0 to n, the position is somewhere between the left and right bounds of that index, then scaled to a percentage (or ratio) of the total

Reply:
`floorTo` is a little confusing also I think. Looking at the context and how this is used, I think what we want to do is take a fraction, some number between 0 and 1, and turn it into a percent, with no decimal.

Could we simplify the `floorTo` function into?
```js
const percent = (value) => Math.trunc(value*100)
```

and would simplify evenlyDistribute, into?
```js
const evenlyDistribute = (index, total) => percent((index+Math.random() * 0.8)/total)
```

the 0.8 still seems odd though, was the idea to avoid the top 20% off each interval? 
Maybe also a nitpick about the name too. "Evenly" do anything and `Math.random` seem at odds with each other.